doc/source/data/faq.rst

```diff
@@ -0,0 +1,308 @@
+  processing.
+* You want to decrease the time-to-first-batch (latency) for a certain operation at the
+  end of your workload. This is the case for training and inference since this prevents
+  GPUs from being idle (which is costly), and can be advantageous for some other
+  latency-sensitive consumers of datasets.
+
+When should I use global per-epoch shuffling?
+=============================================
+
+Global per-epoch shuffling should only be used if your model is sensitive to the
+randomness of the training data. The current status quo is typically to have a per-shard
+in-memory shuffle buffer that you periodically pop random batches from, without mixing
+data across shards between epochs. There is
+`theoretical foundation <https://arxiv.org/abs/1709.10432>`__ for all
+gradient-descent-based model trainers benefiting from improved shuffle quality, and we've
+found that this is particular pronounced for tabular data/models in practice. However,
+the more global your shuffle is, the expensive the shuffling operation, and this
+compounds when doing distributed data-parallel training on a multi-node cluster due to
+data transfer costs, and this cost can be prohibitive when using very large datasets.
+
+The best route for determining the best tradeoff between preprocessing time + cost and
+per-epoch shuffle quality is to measure the precision gain per training step for your
+particular model under different shuffling policies:
+
+* no shuffling,
+* local (per-shard) limited-memory shuffle buffer,
+* local (per-shard) shuffling,
+* windowed (psuedo-global) shuffling, and
+* fully global shuffling.
+
+From the perspective of keeping preprocessing time in check, as long as your data
+loading + shuffling throughput is higher than your training throughput, your GPU should
+be saturated, so we like to recommend users with shuffle-sensitive models to push their
+shuffle quality higher until this threshold is hit.
+
+What is Arrow and how does Ray Datasets use it?
+===============================================
+
+`Apache Arrow <https://arrow.apache.org/>`__ is a columnar memory format and a
+single-node data processing and I/O library that Ray Datasets leverages extensively. You
+can think of Ray Datasets as orchestrating distributed processing of Arrow data.
+
+See our :ref:`key concepts <data_key_concepts>` for more information on how Ray Datasets
+uses Arrow.
+
+How much performance tuning does Ray Datasets require?
+======================================================
+
+Ray Datasets doesn't perform query optimization, so some manual performance
+tuning may be necessary depending on your use case and data scale. Please see our
+:ref:`performance tuning guide <data_performance_tips>` for more information.
+
+How can I contribute to Ray Datasets?
+=====================================
+
+We're always happy to accept external contributions! If you have a question, a feature
```

Comment:
Maybe also link to the docs contrib guide? https://docs.ray.io/en/latest/ray-contribute/docs.html